napari/layers/_image_layer/model.py

```diff
@@ -173,38 +173,10 @@ def _get_shape(self):
             return self.image.shape[:-1]
         return self.image.shape
 
-    def _update(self):
-        """Update the underlying visual.
-        """
-        if self._need_display_update:
-            self._need_display_update = False
-
-            self._node._need_colortransform_update = True
-            self._set_view_slice(self.viewer.dims.indices)
-
-        if self._need_visual_update:
-            self._need_visual_update = False
-            self._node.update()
-
-    def _refresh(self):
-        """Fully refresh the underlying visual.
-        """
-        self._need_display_update = True
-        self._update()
+    def _slice_image(self):
+        """Determines the slice of image from the indices."""
 
-    def _slice_image(self, indices):
-        """Determines the slice of image given the indices.
-
-        Parameters
-        ----------
-        indices : sequence of int or slice
-            Indices to slice with.
-        """
-
-        ndim = self.ndim
-
-        indices = list(indices)
-        indices = indices[-ndim:]
+        indices = list(self.indices)
```

Comment:
I'm confused about why the `ndim` slicing is no longer needed, and ditto for the indices argument. It might be more prudent to give indices a default value of `None` and then write:

```python
indices = indices or self.indices
```

Reply:
the ndim slicing was moved directly onto the base layer. I'm not really sure I see a need for passing custom indices here, though we could always add that in